# MMG 425 HOMEWORK ASSIGNMENT, SPRING SEMESTER 2019

## LEARNING OBJECTIVE

1. To acquaint MMG 425 participants with the process of microbial community analysis of 16S rRNA gene sequence data using RDP's pipeline and community comparison systems.
2. To familiarize MMG 425 participants with FastA files and the Ribosomal Database II Project (RDP II) Classifier.
3. To classify communities into bacterial taxa. 
4. To construct: (a) Sampling Curve and (b) Rank Abundance Curve figures from classified FastA 16S rRNA gene sequence files.
5. To compare bacterial community composition of different ecosystems.

## MATERIALS
```
1. Raw 16S rRNA gene sequence data set from coal mine fire affected soil samples of Centralia.
2. Access to the Ribosomal Database II Project website. 
3. Laptops with wireless connection capability.
4. Spreadsheet and graphics software such as Excel.
```
### FastA Files
```
For FastA file structure, read https://submit.ncbi.nlm.nih.gov/genbank/help/ 
and http://en.wikipedia.org/wiki/FASTA_format

FastA files are simple text files containing a descriptor of a nucleotide 
or amino acid sequence set apart by a ">" sign and a name of the sequence, followed on the next line by the sequence itself. There must be carriage returns at the end of each line, and each line may not be more than 80 characters in length.

Example of two 16S rDNA nucleotide sequences
in the FastA format:
>OTU_dn_1
TACGTAGGGGGCGAGCGTTGTCCGGAGTTACTGGGCGTAAAGCGCACGCAGGCGGCCCACCAAGTTTGAGGTGACAGTCTCCGGCTTAACTGGGGAAGGACCTTGAAGACTGGGGGGCTTGAGGACTTGAGAGGGACACGGAATTCCGGGTGTAGTGGTGAAATGCGTAGATATCCGGAGGAACACCAAAGGCGAAGGCAGTGTCCTGGCAAGCTCCTGACGCTCAGGTGCGAAAGCCAGGGGAGCGAACGGG
>OTU_dn_2
TACGTAGGGGGCGAGCGTTGTCCGGAGTTACTGGGCGTAAAGCGCACGCAGGCGGTCGCCCGCGTTTCGGCTGACAGTGGAGGGCTTCACTCTCCAAGGCGTCGAAAAACGGGGTGACTGGAGGGCCTCAGAGGGACCTGGAACTCCGGGTGGAGCGGTGAAATGCGTAGAGATCCGGAAGAACACCACAGGCGAAGGCAGGGTCCTGGGAGGTACCTGACGCTCAGGTGCGAAAGCAAGGGGAGCGAACGGG
```
## TUTORIAL FOR 16S rRNA GENE SEQUENCES WORKFLOW USING RDP's PIPELINE
```
Form into CATME teams and gather together in the class room.  
Have at least one laptop with wireless connection amongst you,
with Excel and a web browser open.
```
### 1. Download the 16S rRNA gene sequence data on D2L
```
We will use paired-end of 16S rRNA gene sequence reads generated by Ilumina Sequencing tool. There are three pairs of fastQ files (six sequence fastQ files in total) that were compressed into one .tar file. 

For example: 
C01_05102014_R1_D01_GGAGACAAGGGA_L001_R1_001.fastq 
C01_05102014_R1_D01_GGAGACAAGGGA_L001_R2_001.fastq
```
### 2. Go to http://rdp.cme.msu.edu/tutorials/workflows/16S_supervised_flow.html to observe the 16S supervised workflow of RDP's pipeline 
```
The amplicon analysis of 16S rRNA gene sequences is needed to investigate complex bacterial/archaeal communities. There are several pipelines that can be used for amplicon analysis such as Quantitative Insights into Microbial Ecology (QIIME) (Caporaso et al. 2010), mothur (Schloss et al. 2009), and the RDPipeline (Ribosomal Database Project Pipeline) (Cole et al. 2014).
To get more info of the RDPipeline, you can go to the website:
http://pyro.cme.msu.edu
```
### 3. The RDPipeline Processing Steps:

#### Step 1. Initial Processing (Assemble Paired End Reads)
```
The initial processing of RDPipeline contains multi-steps process including sorting the raw reads by sample tag, trimming off tag and primer regions, and removing low quality sequences (Cole et al. 2014). It also uses a tool called Assembler to assembly the paired-end reads (Cole et al. 2014).

1.) Use the fastQ files of 16S rRNA gene sequence data that have been downloaded as the input FASTQ files
2.) Go to https://pyro.cme.msu.edu/login.spr (you don't need to login or make an account) and hit "TEST DRIVE" button.
3.) Upload the .tar file to the initial processing tool main page: https://pyro.cme.msu.edu/init/form.spr
4.) Fill the initial processing form. The forward primer(s) or reverse primer(s) are not required. 
Job name = use any name
Max number of N's = 0
Min Read Q score = 27
Min sequence length = 250
Max sequence length = 280
5.) Make sure to check the box "Assemble paired end reads". 
6.) Hit "Perform Initial Processing" button and wait for a while.
7.) Check if your job is still running or complete by hitting "my jobs" tab on the right corner of the window.
8.) Download the output file once the job is complete.
9.) Inspect your output data by looking at the output file: http://rdp.cme.msu.edu/tutorials/init_process/RDPtutorial_INITIAL-PROCESS_pe.html
```
#### Step 2. Chimera Removal
```
```
#### Step 3. Taxonomic Classification Using RDP Classifier
```
1.) Retrieve the FastA files from the D2L course site. 
2.) Open the RDP Classifier at https://pyro.cme.msu.edu/classifier/form.spr and inspect it.
Toggle on the Classifier tab and inspect it. 
3.) “Run” the classifier with each of the three files by attaching the text file as the procedure indicates, and uploading. Wait for the output.  
4.) Examine the output for dominant and rare taxa. 
5.) ANSWER THIS QUESTION: what is the most common taxon at the genus level?  
6.) Capture the output in a text file and dump it to Excel.  
7.) Sort the file to remove the “zero” values.  
8.) Sort the file again by “genus” to capture the frequency of matches to genus. 
9.) SAVE THE THREE CLASSIFICATIONS IN AN EXCEL FILE USING GENUS AS THE IDENTIFIER.  DISCARD ALL OTHER IDENTIFIERS (FAMILY, ORDER, SPECIES, ETC.).
```
### RANK ABUNDANCE CURVE: EVENNESS
```  
1.) Construct a Rank Abundance Curve of genus x frequency of match using the histogram graph function in Excel. 
2.) Use only genus classified results.  
3.) Repeat and compare for the remaining three FastA files. 
4.) SAVE THE THREE CURVES IN AN EXCEL FILE.
```
### SAMPLING CURVE: ESTIMATION OF RICHNESS
```
(1) Construct a Sampling Curve at the genus level using the following procedure. 
(2) Resample each of the FastA files by taking approximately 40 rows of data at a time and classifying each set of 40 rows. 
(3) Use only genus classified results.  Construct a set of samples (for 371 rows, that will be about 9 samples, maybe 10) into a sampling curve by plotting the number of genera that are discovered on the Y axis and the sample number on the X axis.  
(4) ANSWER THESE QUESTIONS:  Does the curve deflect?  Can you estimate Richness from the curve using the asymptotic method? 
(5) SAVE THE FOUR CURVES IN AN EXCEL FILE.
```
### COMMUNITY COMPARISONS
```
(1) Using the LIBCOMPARE tool in RDP II, submit pairwise FastA files for community composition for the four files.  
There will be 6 comparisons.  
The “significance” test is a probability value, thus 6.1EXP-1 is read as P = 0.61 (i.e., not significant).  
A value of 6.1EXP-3 is read as P = 0.0061 (i.e., a significant difference in frequencies of the taxa of interest). 
(2) ANSWER THESE QUESTIONS: Which of the communities are rather similar and which are very different?  Does “organic” differ from “conventional?”  Does “fly gut” differ from “cow feces?”
```









